lib/statsd.js

```diff
@@ -77,32 +76,39 @@ const Client = function (host, port, prefix, suffix, globalize, cacheDns, mock,
     if (options.dnsError) {
       this.dnsError = options.dnsError;
     }
+    this.socket = options.socket;
```

Comment:
This is taken from what used to be: 
`this.socket = options.isChild ? options.socket : createTransport(this, {`
The creation of the socket is moved up into this if block so that the transport is created at the right time.  This changes the creation of the transport for the cacheDns case to happen after the dns.lookup is done below so that it has the correct host.